airflow/models/baseoperator.py

```diff
@@ -646,7 +646,13 @@ def render_template_from_field(self, attr, content, context, jinja_env):
         rt = self.render_template
         if isinstance(content, str):
             result = jinja_env.from_string(content).render(**context)
-        elif isinstance(content, (list, tuple)):
+        elif isinstance(content, tuple):
+            if type(content) is not tuple:
+                # Special case for named tuples
+                result = content.__class__(*(rt(attr, e, context) for e in content))
+            else:
+                result = (rt(attr, e, context) for e in content)
```

Comment:
This is returning a generator expression, not a tuple :(

```suggestion
                result = tuple(rt(attr, e, context) for e in content)
```

Reply:
Ah, I also need to use a different method for test 52 (since that will never be a list)